### PR TITLE
Implements count-zeroes -- closes #1

### LINKED ISSUE #1: Implement count-zeroes
## Goal

- There is a folder src/count-zeroes. Implement the function count-zeroes in `src/count-zeroes/index.js` and write some unit tests to make sure that your implementation is working as expected across different edge cases
- You can use the `two-plus-two` folder as an example of how to do things.

### Step 1: Setup

- Fork this repository using the "Fork" button on the top right of this page
- Forking will create a new copy of this repository, but on your personal github account
- Clone the local copy you just forked onto your computer
- Setup a remote to the original copy of this repo `PowerNote/Exercise`
  - from your terminal, inside this repo:
  - `git remote add upstream https://github.com/PowerNote/Exercises.git`
- Checkout a new branch to implement this feature: `git checkout -b count-zeroes`
- Go through the README in this repo and follow the setup instructions

### Step 2: Write your tests

- It's good to get in the habit of writing your unit tests before writing the code
  1. This will force you to think about all the edge cases before implementing things
  2. You stay focused on testing the behavior of your function, and not implementation details.
- Not every piece of code you write will have unit tests, and there are also good reasons to write your code before your unit tests, but by default, you should try to write unit tests for your code and you should start by writing those unit tests.
- You should become familiar with jest matchers. A matcher is a statement that makes an assertion about your code. For example:
  - `expect(number).toBe(1);`
  - `expect(number).toBeGreaterThan(7);`
- For a list of matchers that jest supports, you can look [here](https://facebook.github.io/jest/docs/using-matchers.html#content)

### Step 3: Implement the function

- Implement the function and make sure that your unit tests pass
- `npm run test`

### Step 4:  Submit Code

- Commit your code. Use the following commit message:
  - `git commit -m "Implements count-zeroes -- closes #1"`
  - We always have our commit messages in present tense
  - We always indicate the issue we are closing with the syntax `-- closes <number>`
    - By using that syntax, the issue is automatically marked as closed when the code is merged into the master branch
- You now need to submit a pull request so that your code can be merged in with the original repo
- Push your code up to your online repo: `git push origin count-zeroes`
- Go to your online repo, and there should be some dialog that indicates you can submit a pull request for that change. Use the dialog to submit the pull request
- More info on pull requests is [here](https://help.github.com/articles/about-pull-requests/)
  - If your code looks good, I will approve it. If it needs some work, I'll comment on parts of code that you need to adjust and you should go back and commit any changes in order to address those comments. If you push up the new code you committed and it is part of the same feature branch `count-zeroes`, then the code will automatically get added to the pull request
  - Note that when you go back and add followup commits to address changes to the code, you do not need to include `-- closes #1` in the message. Only 1 commit message should indicate the issue being closed

### PR DESCRIPTION
**Pull request by alfigureitout**
Won't take as long to complete in the future. Thank you for your patience.

### CODE REVIEW

File: src/count-zeroes/__tests__/index-tests.js [hunk 1]
```diff
@@ -4,3 +4,19 @@ import countZeroes from '..';
 test('counts zeroes from empty array', () => {
   expect(countZeroes([])).toBe(0);
 });
+
+test('counts zeroes from singleton 0 array', () => {
+  expect(countZeroes([0])).toBe(1);
+});
+
+test('counts zeroes from a straightforward non-zero array', () => {
+  expect(countZeroes([1,2,3,4])).toBe(0);
```
Comment (brenmcnamara) [inline src/count-zeroes/__tests__/index-tests.js]: Should add spaces after your commas: `[1, 2, 3, 4]`

File: src/count-zeroes/__tests__/index-tests.js [hunk 2]
```diff
@@ -4,3 +4,19 @@ import countZeroes from '..';
...
+ });
+
+ test('counts zeroes from an array with 2 zeroes', () => {
+  expect(countZeroes([0,1,2,3,4,0])).toBe(2);
```
Comment (brenmcnamara) [inline src/count-zeroes/__tests__/index-tests.js]: ditto

File: src/count-zeroes/__tests__/index-tests.js [hunk 3]
```diff
@@ -4,3 +4,19 @@ import countZeroes from '..';
...
+ });
+
+ test('counts zeroes from an array with 4 zeroes', () => {
+  expect(countZeroes([0,1,0,1,0,1,0])).toBe(4);
```
Comment (brenmcnamara) [inline src/count-zeroes/__tests__/index-tests.js]: ditto

File: src/count-zeroes/index.js [hunk 4]
```diff
@@ -1,4 +1,11 @@
 
 export default function countZeroes(array) {
-  return 0;
+  return array.map(countIfZero).reduce((prev, curr) => prev + curr, 0);
```
Comment (brenmcnamara) [inline src/count-zeroes/index.js]: map / reduce is overkill. Map loops through the array and creates a new copy, then reduce again loops through the array. You're looping through the array twice to do something you can do in one loop. Just use `reduce`.

```
return array.reduce((sum, num) => num === 0 ? sum + 1 : sum, 0)
```

Comment (alfigureitout) [inline src/count-zeroes/index.js]: Thank you for explanation

**Comment by brenmcnamara** [comment]
Nice!!
